comp/core/status/component_mock.go

```diff
@@ -0,0 +1,13 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build test
+
+package status
+
+// Mock implements mock-specific methods.
+type Mock interface {
+	Component
+}
```

Comment:
Do we need a mock ? Since providers will never use the `status` comp directly they might only test their internal provider.

Reply:
Yeah, that makes total sense. I wanted to have the same structure as the rest of the components. We can remove the mock in the future once we validate no comp test needs a mock 😄

Reply:
Sure, that works. For example the flare we don't have a mock for the comp in itself but we have one for the object provider receive to add data to the current flare.